matter_server/common/models/node.py

```diff
@@ -0,0 +1,133 @@
+"""Matter node."""
+from __future__ import annotations
+from dataclasses import dataclass, field
+from datetime import datetime
+
+from typing import TYPE_CHECKING, Dict
+import logging
+
+from .device_types import RootNode, Aggregator, ALL_TYPES, BridgedDevice
+from chip.clusters import Objects as Clusters, Attribute
+
+from ..helpers.util import dataclass_to_dict
+from .device_type_instance import MatterDeviceTypeInstance
+from .node_device import (
+    AbstractMatterNodeDevice,
+    MatterBridgedNodeDevice,
+    MatterNodeDevice,
+)
+
+from chip.clusters import (
+    Attribute,
+    Cluster,
+    ClusterObject,
+    ClusterAttributeDescriptor,
+    ClusterEvent,
+    ClusterObjects,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class MatterNode:
+    """Matter node."""
+
+    nodeid: int
+    date_commissioned: datetime
+    last_interview: datetime
+    interview_version: int
+    # attributes are stored in form of endpoint: {ClusterID: Cluster}
+    attributes: Dict[int, Dict[int, Cluster]] = field(default_factory=dict)
+    # attributes below will be auto derrived from the attributes
+    # root_device_type_instance: MatterDeviceTypeInstance[RootNode] | None = None
+    # aggregator_device_type_instance: MatterDeviceTypeInstance[Aggregator] | None = None
+    # device_type_instances: list[MatterDeviceTypeInstance] = field(default_factory=list)
+    # node_devices: list[AbstractMatterNodeDevice] = field(default_factory=list)
+
+    def parse_attributes(self, attributes: dict) -> None:
+        """Try to parse a MatterNode from AttributeCache (retrieved from Read)."""
+
+        device_type_instances: list[MatterDeviceTypeInstance] = []
```

Comment:
`device_type_instances` isn't used in this method.

Reply:
WIP, I was working on that now and removed some code by accident, thanks for the reminder ;-)